### PR TITLE
Use ruff instead of black, isort, and flake8

### PR DESCRIPTION
**Pull request by cbourjau**
This PR replaces black, isort, and flake8 with ruff. Ruff does a great job removing unused imports in our opsets. There are some other minor formatting changes that I think are not worth the trouble of diverging from the default settings. 

Supersedes #133

### CODE REVIEW

**Comment by jbachurski** [comment]
Looks good 🚀 
However, I'm not super familiar with the ruff configuration details, maybe someone else can take a look to approve.

**Comment by cbourjau** [comment]
@adityagoel4512 Would you like to do the honors unless you see something amiss? Given that the config is quite defaulty and the diff is reasonably small I'd be happy to just see how it works out.

File: tests/conftest.py [hunk 1]
```diff
@@ -26,7 +26,10 @@ def __init__(self):
         self._last_session = None
 
     def run(self, graph: Graph, unwrap: Optional[str] = None, **kwargs):
-        debug_index = {**graph._get_build_result().scope.var.of_name, **graph._get_build_result().scope.node.of_name}  # type: ignore
+        debug_index = {
+            **graph._get_build_result().scope.var.of_name,
+            **graph._get_build_result().scope.node.of_name,
+        }  # type: ignore
```
Comment (adityagoel4512) [inline tests/conftest.py:32]:
```suggestion
        }
```

**Review by adityagoel4512 — APPROVED** [review]
Not really in the scope of this PR but it would be great if we could address the attribute reference type hinting (e.g. by updating the opset generation).

**Comment by cbourjau** [comment]
The attribute references are still private which is the reason why they are not in the public interface. On the other hand, maybe it is time to stabilize ONNX functions in Spox now that their support in the onnxruntime seems to have matured (even though I consider them pretty broken and useless in the standard).

**Pull request merged**